### PR TITLE
MLPAB-1452 Enable pay on pre/prod and move payment created event

### PR DESCRIPTION
**Pull request by hawx**

### CODE REVIEW

File: cmd/mlpa/main.go [hunk 1]
```diff
@@ -97,6 +97,7 @@ func run(ctx context.Context, logger *slog.Logger) error {
 		notifyIsProduction    = env.Get("GOVUK_NOTIFY_IS_PRODUCTION", "") == "1"
 		ordnanceSurveyBaseURL = env.Get("ORDNANCE_SURVEY_BASE_URL", "http://mock-os-api:8080")
 		payBaseURL            = env.Get("GOVUK_PAY_BASE_URL", "http://mock-pay:8080")
+		payEnabled            = os.Getenv("GOVUK_PAY_ENABLED") == "1"
```
Comment (hawx) [inline cmd/mlpa/main.go:100]: I thought this would be easier than pushing and deploying mock-pay

File: internal/pay/client.go [hunk 2]
```diff
@@ -8,10 +8,13 @@ import (
 	"io"
 	"log/slog"
 	"net/http"
+	"regexp"
 
 	"github.com/ministryofjustice/opg-modernising-lpa/internal/event"
 )
 
+var paymentsURLRe = regexp.MustCompile(`https://[a-z]+\.payments\.service\.gov\.uk/.+`)
```
Comment (hawx) [inline internal/pay/client.go:16]: turns out it was enabled, we just were ignoring the redirects which would go to `card.payments.service.gov.uk`. I've made it more flexible to handle possible future changes

File: internal/page/donor/payment_confirmation.go [hunk 3]
```diff
@@ -35,13 +37,25 @@ func PaymentConfirmation(logger Logger, tmpl template.Template, payClient PayCli
 			return fmt.Errorf("unable to retrieve payment info: %w", err)
 		}
 
+		if payment.State.Status != "success" {
+			return errors.New("TODO: we need to give some options")
```
Comment (hawx) [inline internal/page/donor/payment_confirmation.go:41]: I've added https://opgtransform.atlassian.net/browse/MLPAB-2182 to look at this

**Review by andrewpearce-digital — APPROVED** [review]
infra code looks good

**Review by acsauk — APPROVED** [review]
:shipit:

**Pull request merged**